### PR TITLE
Remove dep on stylelint-rules

### LINKED ISSUE #2689: Ensure we're linting all SCSS files
...currently, we're only going one level deep.

### PR DESCRIPTION
**Pull request by el-mapache**
## Description

Use `gulp-stylelint` directly and port over the non-custom rules from `18f/stylelint-rules`.

Fixes issue with ignored files getting logged to the console

Fixes #2689.

### CODE REVIEW

**Comment by el-mapache** [comment]
@maya, @thisisdano I'd love if one or both of ya'll could look at the scss linting failures on circle and see if the rules producing them are ones we want to follow, or ones we should disable. Thanks!

**Comment by maya** [comment]
@el-mapache took a look at this, from what I can tell, those rules look good to follow, is it something we can easily fix with autofix? https://stylelint.io/user-guide/cli/#autofixing-errors or @thisisdano did you want to take a look at cleaning these up?

**Comment by el-mapache** [comment]
@maya hopefully! Thanks for taking a look, I'll try the autofix tool. Shouldn't take too long to manually resolve these though, if necessary.

**Comment by thisisdano** [comment]
Sorry, I've been in a bit of a hole — looking at this now

**Comment by thisisdano** [comment]
All these rules seem good — _maybe_ with the exception of `block-closing-brace-newline-after`. I'm happy to follow it, but it's the difference between

```sass
  @if map-has-key($our-breakpoints, $quoted-bp){
    @if $theme-respect-user-font-size {
      $bp: rem-to-user-em(map-get($our-breakpoints, $quoted-bp));
    } @else {
      $bp: rem-to-px(map-get($our-breakpoints, $quoted-bp));
    }
  }
``` 

and

```sass
  @if map-has-key($our-breakpoints, $quoted-bp){
    @if $theme-respect-user-font-size {
      $bp: rem-to-user-em(map-get($our-breakpoints, $quoted-bp));
    } 
    @else {
      $bp: rem-to-px(map-get($our-breakpoints, $quoted-bp));
    }
  }
```

It's a bit _de gustibus_ though I'd agree the the second one is a bit easier to read, if a little loose.

**Comment by maya** [comment]
Can we keep this rule but ignore for `else` statements? https://stylelint.io/user-guide/rules/block-closing-brace-newline-after/#optional-secondary-options

File: src/stylesheets/core/mixins/_layout-grid.scss [hunk 1]
```diff
@@ -42,7 +42,7 @@
     @elseif map-has-key($uswds-column-gaps, $gap) {
```
Comment (maya) [inline src/stylesheets/core/mixins/_layout-grid.scss]: @thisisdano shouldn't this be `else if` not `elseif`? http://sass-lang.com/documentation/file.SASS_REFERENCE.html#control_directives__expressions

Comment (thisisdano) [reply src/stylesheets/core/mixins/_layout-grid.scss]: Yeah, I've been using `esleifs`, but I can find change to `else if`

File: src/stylesheets/core/mixins/_at-media.scss [hunk 2]
```diff
@@ -3,15 +3,16 @@
   $orig-bp: $bp;
   $quoted-bp: quote($bp);
   $our-breakpoints: map-deep-get($system-properties, breakpoints, standard);
-  @if map-has-key($our-breakpoints, $quoted-bp){
+  @if map-has-key($our-breakpoints, $quoted-bp) {
     @if $theme-respect-user-font-size {
       $bp: rem-to-user-em(map-get($our-breakpoints, $quoted-bp));
-    } @else {
+    }
+    @else {
```
Comment (maya) [inline src/stylesheets/core/mixins/_at-media.scss]: @uswds/core Since we disable this for `@else`, do we want to put these back to how they were?

Comment (maya) [reply src/stylesheets/core/mixins/_at-media.scss]: or should we leave them for consistency, looking through some instances here, I see we wrote them both ways.

File: src/stylesheets/core/mixins/_layout-grid.scss (hunk 1 above)
Comment (maya) [reply src/stylesheets/core/mixins/_layout-grid.scss]: @thisisdano in this PR or a separate one?

File: src/stylesheets/core/mixins/_at-media.scss (hunk 2 above)
Comment (thisisdano) [reply src/stylesheets/core/mixins/_at-media.scss]: Maybe I should work on making them consistent in a subsequent PR

Comment (el-mapache) [reply src/stylesheets/core/mixins/_at-media.scss]: Doesn't matter to me, I think we have a lot more cases of having the `else`s on another line, for what it's worth

Comment (thisisdano) [reply src/stylesheets/core/mixins/_at-media.scss]: And call this work done

Comment (maya) [reply src/stylesheets/core/mixins/_at-media.scss]: That works for me, approved!

**Pull request merged**